src/classLibrary/ParsingStrategies/InlineFlagResolver.ts

```diff
@@ -0,0 +1,174 @@
+            },
+            default: () => false,
+        });
+	}
+
+    ShouldKeepStringWithInlineFlag(flagSignatures: string[], flagValue: boolean, checkTarget: string): boolean {
+        return flagSignatures.Where((signature: string) => !checkTarget.includes(signature) || flagValue).length === flagSignatures.length;   
+    }
+
+    RemoveFlagFromMetadataDestinationPath(signatures: string[], virtualFile: VirtualFileSystemInstance) {
+        VirtualFileSystemInstance.match(virtualFile, {
+            File: (file: FileSystemInstance) => {
+                file.metadata.destinationAbsolutePath = file.metadata.destinationAbsolutePath.Without(signatures.Map((s: string) => s));
+            },
+            Folder: (folder: DirectorySystemInstance) => {
+                folder.metadata.destinationAbsolutePath = folder.metadata.destinationAbsolutePath.Without(signatures.Map((s: string) => s));
+            },
+        });
+	};
+
+	ResolveContents(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): VirtualFileSystemInstance[] {
+        const flagsMap: Map<string, boolean> = this.util.FlattenObject(cyan.flags).MapValue((boolString: string) => boolString == "true");
+        const allPossibleFlagsMap: Map<string[], boolean> = flagsMap.MapKey((key: string) => this.ModifyFlagWithAllSyntax(key, cyan.syntax));
+        const allPossibleInverseFlagsMap: Map<string[], boolean> = flagsMap.MapKey((key: string) => this.ModifyInverseFlagWithAllSyntax(key, cyan.syntax));
+
+        return virtualFiles.Each((virtualFile: VirtualFileSystemInstance) => {
+            VirtualFileSystemInstance.match(virtualFile, {
+                File: (file: FileSystemInstance) => {
+                    if (!file.ignore.inlineResolver.content) return;
+                    if (file["content"] == null) return;
+                    FileContent.if.String(file.content, (str: string) => {
+                        str = this.ConstructContentForInlineFlags(str, allPossibleFlagsMap, cyan.comments, false);
+                        str = this.ConstructContentForInlineFlags(str, allPossibleInverseFlagsMap, cyan.comments, true);
+                        file.content = FileContent.String(str);
+                    });
+                },
+                default: () => {
+                    return;
+                }
+            });
+        });
+	};
+
+    ConstructContentForInlineFlags(content: string, allPossibleSignatureMap: Map<string[], boolean>, comments: string[], isInverse: boolean): string {
+        allPossibleSignatureMap
+            .Each((allSignatures: string[], val: boolean) => {
+                if (isInverse) val = !val;
+                content = content
+                    .LineBreak()
+                    .Where((s: string) => this.ShouldKeepStringWithInlineFlag(allSignatures, val, s))
+                    .Map((s: string) => s.Without(this.GenerateCommentsWithSignatureStrings(allSignatures, comments))) //why should it care about comments?
```

Comment:
I think the main reason why it should care about comments is because most in-line flags will appear like this:

```
code here // flag~condition~
```

so the inlineFlagResolver should remove the comments if the condition is true